test/router/path_trie_crud_test.dart

```diff
@@ -175,6 +175,61 @@ void main() {
         expect(() => trie.update(pathDefinition, 1), throwsArgumentError);
         expect(trie.lookup(NormalizedPath('/articles/any')), isNull);
       });
+
+      test(
+          'Given a trie with an existing wildcard path /data/* and value, '
+          'when update is called for /data/* with a new value, '
+          'then lookup for a matching path returns the new value', () {
+        final pathDefinition = NormalizedPath('/data/*');
+        trie.add(pathDefinition, 1);
+
+        trie.update(pathDefinition, 2);
```

Comment:
Q: We does this discern between `add` and `update` compared to a just a "set" on a `Map`?

Reply:
Yes, there are `add`, `update`, and `addOrUpdate` with the expected semantics. The first two will throw, if a value is already attached (`add`) , or missing (`update`).